### PR TITLE
Corrections in MO to fix performance drop in Interpolate-4 when mode=linear instead of linear_onnx

### PR DESCRIPTION
**Pull request by vgavrilo**
Corrections in MO to fix performance drop in Interpolate-4 when `mode=linear` instead of `linear_onnx`.
Ticket: #49045

Code:

* [x] Comments
* [x] Code style (PEP8)
* [x] Transformation generates reshape-able IR: Yes
* [x] Transformation preserves node names: Yes

Validation: 

* [x] Unit tests
* [x] Framework layer tests:  N/A because there are no new supported layers
* [x] Transformation tests: N/A
* [x] e2e model test: N/A because there is no new model
* [x] MO IR Reader tests: N/A because there is no new model

Documentation:

* [x] Supported frameworks layers list: N/A because there are no new operations
* [x] Supported **public** models list: N/A because there is no new **public** model
* [x] New operations specification: N/A because there are no new operations
* [x] Guide on how to convert the **public** model: N/A because there is no new **public** model
* [x] User guide update: N/A because there is no need to update common user guide

Other:

* [x] Sample/Demo application to infer the model: N/A because there is now new model

### CODE REVIEW

**Comment by vgavrilo** [comment]
@iimironov, @achetver, @evolosen, @lazarevevgeny, please, see this PR (https://github.com/openvinotoolkit/openvino/pull/4488)

File: model-optimizer/extensions/back/LinearToLinearONNXReplacer.py
Comment (achetver) [inline model-optimizer/extensions/back/LinearToLinearONNXReplacer.py:36]: nit: please, add message with some details to this assert.

**Review by achetver — APPROVED** [review]
Changes looks good.

File: model-optimizer/extensions/back/LinearToLinearONNXReplacer.py
Comment (vgavrilo) [reply model-optimizer/extensions/back/LinearToLinearONNXReplacer.py]: Done.

**Comment by vgavrilo** [comment]
@iimironov, @evolosen, @lazarevevgeny, please, see this PR (#4488)

**Review by lazarevevgeny — APPROVED** [review]
Looks good. Waiting for benchmark run results before merge.

**Review by evolosen — APPROVED** [review]

**Pull request merged**
